internal/pkg/plan/delete-template/plan.go

```diff
@@ -0,0 +1,65 @@
+package delete_template
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/diff"
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/state"
+	"github.com/keboola/keboola-as-code/internal/pkg/state/local"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils"
+)
+
+type DeleteAction struct {
+	State    model.ObjectState
+	Manifest model.ObjectManifest
+}
+
+type Plan struct {
+	actions      []DeleteAction
+	projectState *state.State
+	branchKey    model.BranchKey
+	instanceId   string
+}
+
+func (p *Plan) Empty() bool {
+	return len(p.actions) == 0
+}
+
+func (p *Plan) Name() string {
+	return "delete-template"
+}
+
+func (p *Plan) Log(log log.Logger) {
+	writer := log.InfoWriter()
+	writer.WriteString(fmt.Sprintf(`Plan for "%s" operation:`, p.Name()))
+	actions := p.actions
+
+	if len(actions) == 0 {
+		writer.WriteStringIndent(1, "nothing to delete")
+	} else {
+		for _, action := range actions {
+			writer.WriteStringIndent(1, fmt.Sprintf("%s %s %s", diff.DeleteMark, model.ConfigAbbr, action.State.Path()))
+		}
+	}
+}
+
+func (p *Plan) Invoke(ctx context.Context, localManager *local.Manager) error {
+	if err := newExecutor(ctx, localManager, p).invoke(); err != nil {
+		return err
+	}
+
+	branchState := p.projectState.GetOrNil(p.branchKey).(*model.BranchState)
+	if err := branchState.Local.Metadata.DeleteTemplateUsage(p.instanceId); err != nil {
+		return utils.PrefixError(`cannot remove template instance metadata`, err)
+	}
+	saveOp := p.projectState.LocalManager().NewUnitOfWork(p.projectState.Ctx())
+	saveOp.SaveObject(branchState, branchState.LocalState(), model.NewChangedFields())
+	if err := saveOp.Invoke(); err != nil {
+		return err
+	}
```

Comment:
Toto by sa dalo este cele presunut do `executor.invoke` nie? Nech je to pokope, myslim, ze aj u ostatnych je `plan.Invoke` iba skratka nad `newExecutor.invoke`.

Reply:
Jo ok. Moc architektuře toho Plánu nerozumím, tak jsem to dal kam mi přišlo nejlepší. 😄

Reply:
Mozeme to prebrat na standup ak mas nejake otazky, .... cele je to len o tom aby bolo generovanie planu a jeho vykonanie oddelene. preto je tam `plan`  a `executor`.

Reply:
Jasně. Jó už vím, nebyl jsem si jistej jestli se nastavení metadat "počítá" do té exekuce samotné, tak asi proto jsem to do ní nedal. Nebo tak nějak. 🤔🙂